Assigning the member role to new members in the Learn English server

Eaky0 opened the pull request:
## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [x] All of the code is easily understood.
- [x] There is no redundant or duplicate code.

## Describe your changes
- **JoinListener**: Added a new class within the bot.listener package. Currently, It only assigns the 'Member' role for new members who have just joined the server. Note: I couldn't test this functionality, as I have no clue on how to use the docker container. 
I have tried this command ```docker run -it -p 27017:27017 --name mongoContainer mongo:latest```, but it didn't work.
- **App**: I added a new JoinListener object in the listeners list in the App class (where the main function resides). 

## Issue number
Atakan here, no Idea what the issue number for this, is supposed to be. I couldn't find this anywhere for this specific problem 😅.

Eaky0 committed:
<ADDED by Eaky0>: JoinListener: A class responsible for detecting joining members and assigning them the member role!

Eaky0 committed:
<MODIFIED by Eaky0>: JoinListener: Modified to the right member role id

Eaky0 committed:
<MODIFIED by Eaky0>: Forgot to do gradlew spotlessApply

christolis commented on app/src/main/java/bot/listener/JoinListener.java:
```diff
@@ -0,0 +1,36 @@
+        try {
+            // 505348918520053760L Member role id in the Official Learn English Server.
+            // 1120239610920976563L Member role id in the Learn English (Staging) Server.
+            Role role = event.getGuild().getRoleById(505348918520053760L);
```
Let's take this long number and put it as a static variable in this class

christolis commented on app/src/main/java/bot/listener/JoinListener.java:
```diff
@@ -0,0 +1,36 @@
+            }
+
+        } catch (NullPointerException e) {
+            System.out.println("This role doesn't exist!");
```
Let's use a logger for this :)

Eaky0 committed:
log calls added to joinlistener and static variable member role id defined in constants!

christolis committed:
feat: do not add the member role to one who has it

We do not need to send a request to the Discord API to add the role to
every joining member unconditionally. While at the end, the user will
not receive the role two times, we should still try not to unnecessarily
oversaturate the API with reduntant calls.

christolis commented:
Made a few final adjustments to the `JoinListener` class. Even though one of our bots in the server has the functionality that this pull request implements, we will still merge it, even though it took a great amount of time to do so.

Please refrain from using your git username on branch names, commit messages, etc. Your contributions are already accurately credited per line of code that you have written. Each commit that you create is attributed to your username and e-mail address. Your commit messages should accurately reflect the changes made in your commit in a concise and intuitive manner. At your leisure, check out the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) website which has some useful information about structuring good commit messages.

With all that being said, thank you for your contributions.
